packages/bot/src/scratch/blocks/Text/text_getSubstring.js

```diff
@@ -138,47 +138,66 @@ Blockly.JavaScript.text_getSubstring = block => {
         text.match(/^'?\w+'?$/) ||
         (where1 !== 'FROM_END' && where1 !== 'LAST' && where2 !== 'FROM_END' && where2 !== 'LAST')
     ) {
-        if (where1 === 'FROM_START') {
-            at1 = Blockly.JavaScript.getAdjusted(block, 'AT1');
-        } else if (where1 === 'FROM_END') {
-            at1 = Blockly.JavaScript.getAdjusted(block, 'AT1', 1, false, Blockly.JavaScript.ORDER_SUBTRACTION);
-            at1 = `${text}.length - ${at1}`;
-        } else if (where1 === 'FIRST') {
-            at1 = '0';
+        switch (where1) {
+            case ('FROM_START'): {
```

Comment:
@aaron-binary its better if we start using constants instead of hardcoding. 
we can define a code style, like use some shared constants between blocks or define constants separately in each block definition.